vmware-event-router/internal/processor/aws_event_bridge.go

```diff
@@ -171,102 +178,70 @@ func NewAWSEventBridgeProcessor(ctx context.Context, cfg connection.Config, sour
 	return &eventBridge, nil
 }
 
-// Process implements the stream processor interface TODO: handle
-// throttling/batching
-// https://docs.aws.amazon.com/eventbridge/latest/userguide/cloudwatch-limits-eventbridge.html#putevents-limits
-func (awsEventBridge *awsEventBridgeProcessor) Process(moref types.ManagedObjectReference, baseEvent []types.BaseEvent) error {
-	batchInput, err := awsEventBridge.createPutEventsInput(baseEvent)
-	if err != nil {
-		errMsg := fmt.Errorf("could not create PutEventsInput for event(s): %v", err)
-		awsEventBridge.Println(errMsg)
-		return processorError(ProviderAWS, errMsg)
-	}
-
-	// nothing to send
-	if len(batchInput) == 0 {
-		return nil
+// Process implements the stream processor interface
+func (awsEventBridge *awsEventBridgeProcessor) Process(ce cloudevents.Event) error {
+	if awsEventBridge.verbose {
+		awsEventBridge.Printf("processing event (ID %s): %v", ce.ID(), ce)
 	}
 
-	for idx, input := range batchInput {
-		resp, err := awsEventBridge.PutEvents(&input)
-		if err != nil {
-			awsEventBridge.Printf("could not send event(s) for batch %d: %v", idx, err)
-			continue
-		}
+	awsEventBridge.mu.RLock()
+	defer awsEventBridge.mu.RUnlock()
+	if _, ok := awsEventBridge.patternMap[ce.Subject()]; !ok {
+		// no event bridge rule pattern (subscription) for event, skip
 		if awsEventBridge.verbose {
-			awsEventBridge.Printf("successfully sent event(s) from source %s: %+v batch: %d",
-				awsEventBridge.source,
-				resp,
-				idx)
+			awsEventBridge.Printf("pattern rule does not match, skipping event (ID %s): %v", ce.ID(), ce)
 		}
+		return nil
 	}
-	return nil
-}
-
-func (awsEventBridge *awsEventBridgeProcessor) createPutEventsInput(baseEvent []types.BaseEvent) ([]eventbridge.PutEventsInput, error) {
-	awsEventBridge.mu.Lock()
-	defer awsEventBridge.mu.Unlock()
 
-	batch := []eventbridge.PutEventsInput{}
-
-	tmpInput := eventbridge.PutEventsInput{
-		Entries: []*eventbridge.PutEventsRequestEntry{},
+	jsonBytes, err := json.Marshal(ce)
+	if err != nil {
+		msg := fmt.Errorf("could not marshal event %v: %v", ce, err)
+		awsEventBridge.Println(msg)
```

Comment:
Yeah, I also thought this would be too much since we bubble the error up. Left it in since it's a processor error and currently we don't have providers and processors decoupled. So logging this error at stream provider level (e.g. vCenter) seemed a bit misleading to me. What do you think?

Reply:
Yeah in that context it seems reasonable to leave this as is for now I just saw in `vcenter_fake`:
```
				if err != nil {
					f.Logger.Printf("could not process event %v: %v", ce, err)
				}
```
And though we would just print it twice if there is an error. Your comment is valid so let's leave this as is